transaction_items.go

```diff
@@ -52,3 +54,50 @@ func (t *Client) TransactItems(ctx context.Context, input ...types.TransactWrite
 	})
 	return err
 }
+
+func (t *Client) WithUpdateItem(pk string, sk string, updates map[string]Attribute, opts ...UpdateOption) types.TransactWriteItem {
+	var setExpressions []string
+	expressionAttributeNames := make(map[string]string)
+	expressionAttributeValues := make(map[string]Attribute)
+
+	for key, value := range updates {
+		attrName := fmt.Sprintf("#%s", key)
+		attrValue := fmt.Sprintf(":%s", key)
+		
+		setExpressions = append(setExpressions, fmt.Sprintf("%s = %s", attrName, attrValue))
+		expressionAttributeNames[attrName] = key
+		expressionAttributeValues[attrValue] = value
+	}
+
+	updateExpression := fmt.Sprintf("SET %s", strings.Join(setExpressions, ", "))
+
+	input := &dynamodb.UpdateItemInput{
+		TableName:                 &t.TableName,
+		Key:                       map[string]types.AttributeValue{
+			"pk": &types.AttributeValueMemberS{Value: pk},
+			"sk": &types.AttributeValueMemberS{Value: sk},
+		},
+		UpdateExpression:          &updateExpression,
+		ExpressionAttributeNames:  expressionAttributeNames,
+		ExpressionAttributeValues: expressionAttributeValues,
+	}
+
+	for _, opt := range opts {
+		err := opt(input)
+		if err != nil {
+			log.Println("Failed to apply update option: " + err.Error())
+			return types.TransactWriteItem{}
+		}
```

Comment:
This is problematic. If update option fails to apply, should we proceed with the db update?

Reply:
Removing the log doesn't make any difference. We either need to panic or change TransactWriteOption to return an error